platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java

```diff
@@ -116,8 +117,12 @@ public void loadReconnectState(@NonNull final SignedState signedState) {
                                 + signedState.getState().getHash());
             }
 
-            // Before attempting to load the state, verify that the platform AB matches the state AB.
-            AddressBookUtils.verifyReconnectAddressBooks(addressBook, signedState.getAddressBook());
+            // Before attempting to load the state, verify that the platform roster matches the state roster.
+            final State state = (MerkleStateRoot<?>) signedState.getState().getSwirldState();
+            final Roster stateRoster = RosterRetriever.retrieveActiveOrGenesisRoster(state);
+            if (!roster.equals(stateRoster)) {
+                throw new IllegalStateException("Current roster and state-based roster do not contain the same nodes");
```

Comment:
Minor: I'd still suggest to include both the state and the local rosters in the exception message using `Roster.JSON.toJSON()`.

Reply:
Done